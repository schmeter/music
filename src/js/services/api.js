import configApi from '../../config/api.json';

export const getApiUrl = (id, params = {}) => {
  let url = null;

  if (configApi[id]) {
    url = configApi[id];
    Object.entries(params).forEach(([key, param]) => {
      url = url.replace(`:${key}`, param);
    });
  }
  return url;
};
