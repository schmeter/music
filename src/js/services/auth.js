import md5 from 'md5';

import storage from './storage';

export const isAuthenticated = () => {
    return !!storage.get('auth:isLoggedIn');
};

// expiration in seconds
export const authenticate = (credentials, expiration = 60 * 60) => {
    if (credentials.force || md5(credentials.password) === '2f43b42fd833d1e77420a8dae7419000') {
        storage.set('auth:isLoggedIn', true, expiration);
    }
};

export const unauthenticate = () => {
    storage.unset('auth:isLoggedIn');
};
