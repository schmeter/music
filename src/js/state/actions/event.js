import {
    loadLibrary,
} from '../../services/event';

export const requestEventLibraryAction = () => dispatch => {
    const library = loadLibrary();

    dispatch(setLibraryAction(library));
};

export const setLibraryAction = library => ({
    type: 'EVENT_SET_LIBRARY',
    payload: library,
});
