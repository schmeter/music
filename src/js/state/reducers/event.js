export const initialState = {
    library: {
        events: [],
        artists: [],
    },
};

export default (state = {}, action = {}) => {
    switch (action.type) {
        case 'EVENT_SET_LIBRARY':
            return {
                ...state,
                library: action.payload,
            };
        default:
            return state;
    }
};
