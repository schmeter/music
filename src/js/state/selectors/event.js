export const getLibrary = state => state.event.library;

export const getEvents = state =>
    getLibrary(state).events;

