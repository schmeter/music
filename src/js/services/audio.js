import { propOr } from 'ramda';

import AudioLibraryModel from '../models/AudioLibrary';

import audioData from '../../../.tmp/audio.json';

import { isAuthenticated } from './auth';
import storage from './storage';

import { getRandom } from '../util/math';
import { isPropTrueAtIndex } from '../util/array';

export const getLibrary = (showAll = isAuthenticated()) => new AudioLibraryModel(audioData, showAll);

export const getRandomIndex = (tracks) => {
    const filterList = tracks.filter((item) => !item.hidden);

    return tracks.indexOf(filterList[getRandom(0, filterList.length - 1)]);
};

export const getActiveIndex = (tracks) => {
    const activeIndex = tracks.findIndex(track => track.path === loadActiveTrackPath());

    return activeIndex !== -1 ? activeIndex : getRandomIndex(tracks);
};

export const getNextIndex = (activeIndex, tracks) => {
    if (!tracks[activeIndex]) {
        activeIndex = 0;
    }
    let nextIndex = activeIndex + 1;

    if (!tracks[nextIndex]) {
        nextIndex = 0;
    }
    if (
        isPropTrueAtIndex('skip', nextIndex, tracks)
        || (
            isPropTrueAtIndex('hidden', nextIndex, tracks)
            && !isSameAlbum(tracks[activeIndex], tracks[nextIndex])
        )
    ) {
        for (let index = 0; index < tracks.length; index++) {
            if (!tracks[nextIndex]) {
                nextIndex = 0;
            } else if (
                !isPropTrueAtIndex('skip', nextIndex, tracks)
                && !isPropTrueAtIndex('hidden', nextIndex, tracks)
            ) {
                break;
            } else {
                nextIndex += 1;
            }
        }
    }
    return nextIndex;
};

export const isSameAlbum = (track, nextTrack) =>
    (track && nextTrack) &&
    (track.artist && nextTrack.artist && track.album && nextTrack.album) &&
    (track.artist.id === nextTrack.artist.id) && (track.album.id === nextTrack.album.id);

export const getArtist = (library, artistId) => library.artists.find((artist) => artist.id === artistId);

export const getAlbum = (library, artistId, albumId) =>
    propOr([], 'albums', getArtist(library, artistId)).find((album) => album.id === albumId);

export const loadActiveTrackPath = () => storage.get('audio:activeTrackPath');

export const saveActiveTrackPath = (activeTrackPath) => {
    storage.set('audio:activeTrackPath', activeTrackPath);
};
