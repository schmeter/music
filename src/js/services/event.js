import EventLibraryModel from '../models/EventLibrary';
import { data } from '../helpers/aggregate';
import { isAuthenticated } from '../helpers/auth';

export const loadLibrary = (showAll = isAuthenticated()) => new EventLibraryModel(data.event, showAll);

export const getEventId = event => `${event.artist.title} ${event.date} ${event.title}`;
