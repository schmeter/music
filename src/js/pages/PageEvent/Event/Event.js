import React, { useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import { format } from 'date-fns';
import { useParams } from 'react-router-dom';

import Image from '../../../components/Image';
import Link from '../../../components/Link';
import { getApiUrl } from '../../../services/api';
import { getEventId } from '../../../services/event';

const Event = ({
    event,
}) => {
    const link = getApiUrl('googleRoute', {
        from: '',
        to: event.address || event.title,
    });
    const id = getEventId(event);
    const { eventId } = useParams();
    const eventContainer = useRef();

    useEffect(() => {
        if (eventId === id) {
            eventContainer?.current?.scrollIntoView?.();
        }
    }, [eventId, id]);

    return (
        <div
            ref={eventContainer}
            className="event"
        >
            <h2 className="date">
                {format(new Date(event.date), 'dd.MM.yyyy')}
                &nbsp;
                <Link to={link}>
                    {event.title}
                </Link>
            </h2>
            <h3>
                {event.artist.title}
            </h3>
            {event.imgPath && (
                <Link to={link}>
                    <Image
                        src={event.imgPath}
                        alt={event.title}
                    />
                </Link>
            )}
        </div>
    );
};

Event.propTypes = {
    event: PropTypes.object.isRequired,
};

export default Event;
