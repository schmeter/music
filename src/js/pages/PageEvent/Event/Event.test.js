import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Event from './Event';
import { getEventId } from '../../../services/event';

let mockUrlParams;

jest.mock('react-router-dom', () => ({
  useParams: () => (mockUrlParams),
}));
window.HTMLElement.prototype.scrollIntoView = function() {};
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useRef: () => ({
    current: {
      eventContainer: {
        scrollIntoView: () => {},
      },

    },
  }),
}));

describe('Event', () => {
  const props = {
    event: {
      title: 'test',
      date: '2020-02-02',
      artist: {
        title: 'test',
      },
    },
  };

  it('renders correctly with image', () => {
    mockUrlParams = {};

    const component = mount(<Event {...{ ...props, event: { ...props.event, imgPath: 'test' } } } />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders correctly without url params', () => {
    mockUrlParams = {};

    const component = mount(<Event {...props} />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders correctly with correct url params', () => {
    mockUrlParams = {
      eventId: getEventId(props.event),
    };

    const component = mount(<Event {...props} />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders correctly with wrong url params', () => {
    mockUrlParams = {
      eventId: 'test',
    };

    const component = mount(<Event {...props} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
