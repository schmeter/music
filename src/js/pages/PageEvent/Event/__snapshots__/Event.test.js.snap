// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event renders correctly with correct url params 1`] = `
<Event
  event={
    Object {
      "artist": Object {
        "title": "test",
      },
      "date": "2020-02-02",
      "imgPath": "test",
      "title": "test",
    }
  }
>
  <div
    className="event"
  >
    <Link
      to="https://www.google.de/maps/dir//test"
    >
      <a
        href="https://www.google.de/maps/dir//test"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Image
          alt="test"
          src="test"
        >
          <img
            alt="test"
            className="image invisible"
            onError={[Function]}
            onLoad={[Function]}
            src="test"
          />
        </Image>
      </a>
    </Link>
    <h2
      className="date"
    >
      02.02.2020
       
      <Link
        to="https://www.google.de/maps/dir//test"
      >
        <a
          href="https://www.google.de/maps/dir//test"
          rel="noopener noreferrer"
          target="_blank"
        >
          test
        </a>
      </Link>
    </h2>
    <h3>
      test
    </h3>
  </div>
</Event>
`;

exports[`Event renders correctly with wrong url params 1`] = `
<Event
  event={
    Object {
      "artist": Object {
        "title": "test",
      },
      "date": "2020-02-02",
      "imgPath": "test",
      "title": "test",
    }
  }
>
  <div
    className="event"
  >
    <Link
      to="https://www.google.de/maps/dir//test"
    >
      <a
        href="https://www.google.de/maps/dir//test"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Image
          alt="test"
          src="test"
        >
          <img
            alt="test"
            className="image invisible"
            onError={[Function]}
            onLoad={[Function]}
            src="test"
          />
        </Image>
      </a>
    </Link>
    <h2
      className="date"
    >
      02.02.2020
       
      <Link
        to="https://www.google.de/maps/dir//test"
      >
        <a
          href="https://www.google.de/maps/dir//test"
          rel="noopener noreferrer"
          target="_blank"
        >
          test
        </a>
      </Link>
    </h2>
    <h3>
      test
    </h3>
  </div>
</Event>
`;

exports[`Event renders correctly without url params 1`] = `
<Event
  event={
    Object {
      "artist": Object {
        "title": "test",
      },
      "date": "2020-02-02",
      "imgPath": "test",
      "title": "test",
    }
  }
>
  <div
    className="event"
  >
    <Link
      to="https://www.google.de/maps/dir//test"
    >
      <a
        href="https://www.google.de/maps/dir//test"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Image
          alt="test"
          src="test"
        >
          <img
            alt="test"
            className="image invisible"
            onError={[Function]}
            onLoad={[Function]}
            src="test"
          />
        </Image>
      </a>
    </Link>
    <h2
      className="date"
    >
      02.02.2020
       
      <Link
        to="https://www.google.de/maps/dir//test"
      >
        <a
          href="https://www.google.de/maps/dir//test"
          rel="noopener noreferrer"
          target="_blank"
        >
          test
        </a>
      </Link>
    </h2>
    <h3>
      test
    </h3>
  </div>
</Event>
`;
